syntax = "proto3";

package internet;

service Internet {
  rpc OpenConnection (URL) returns (ReturnStatus) {}
  rpc ReadConnectionStream (Identifier) returns (ReturnedReadBuffer) {}
  rpc CloseConnection (Identifier) returns (Success) {}
}

message URL {
  bytes contractKey = 1;
  bytes callId = 2;
  string url = 3;
  bytes session_id = 5;
}

message Identifier {
  bytes contractKey = 1;
  bytes callId = 2;
  uint64 identifier = 3;
  bytes session_id = 5;
}

message Success {
  bool success = 1;
}

message ReturnStatus {
  bool success = 1;
  uint64 identifier = 2;
}

message ReturnedReadBuffer {
  bool success = 1;
  bytes buffer = 2;
}